### PR TITLE
Add integration test for K8s gateway

### PR DESCRIPTION
**Pull request by howardjohn**
This also fixes a couple of bugs that were found during this

### CODE REVIEW

File: pilot/pkg/config/kube/gateway/conversion.go [hunk 1]
```diff
@@ -72,22 +73,16 @@ func findByName(name, namespace string, cfgs []model.Config) *model.Config {
 	return nil
 }
 
-func convertResources(r *KubernetesResources) (IstioResources, error) {
+func convertResources(r *KubernetesResources) IstioResources {
 	result := IstioResources{}
-	gw, routeMap, err := convertGateway(r)
-	if err != nil {
-		return result, fmt.Errorf("failed to convert gateways: %v", err)
```
Comment (howardjohn) [inline pilot/pkg/config/kube/gateway/conversion.go]: motivation here is that returning an error here blocks the entire push pipeline. We don't want that, we want to propagate the errors into Status on the objects, but that isn't implemented yet

File: pilot/pkg/bootstrap/server.go [hunk 2]
```diff
@@ -781,7 +781,7 @@ func (s *Server) initEventHandlers() error {
 			}
 			s.EnvoyXdsServer.ConfigUpdate(pushReq)
 		}
-		for _, schema := range collections.Pilot.All() {
+		for _, schema := range collections.PilotServiceApi.All() {
```
Comment (hzxuzhonghu) [inline pilot/pkg/bootstrap/server.go]: guard with `if features.EnableServiceApis`, consistent with that in `makeKubeConfigController`

Comment (howardjohn) [reply pilot/pkg/bootstrap/server.go]: Ill do that but this part doesn't really matter, it will ignore CRDs we are not watching in the controller

Comment (howardjohn) [reply pilot/pkg/bootstrap/server.go]: I have changed it

File: tests/integration/pilot/ingress/ingress_test.go [hunk 3]
```diff
@@ -70,6 +77,105 @@ func TestMain(m *testing.M) {
...
+  - hostname: "my.domain.example"
+    rules:
+    - match:
+        pathType: Prefix
+        path: /get
+      filter:
+        headers:
+          add:
+            my-added-header: added-value
+          remove: [my-removed-header]
```
Comment (hzxuzhonghu) [inline tests/integration/pilot/ingress/ingress_test.go]: nit: i donot think this is tested

**Review by hzxuzhonghu — APPROVED** [review]
lgtm

File: tests/integration/pilot/ingress/ingress_test.go (hunk 3 above)
Comment (nmittler) [reply tests/integration/pilot/ingress/ingress_test.go]: +1

Comment (howardjohn) [reply tests/integration/pilot/ingress/ingress_test.go]: I'll remove this. I think (at least for now) it makes sense to have this test basic and have the bulk of the detailed coverage as unit tests (under the assumption virtual service integ testing is sufficient, and we can rely on that)

**Review by nmittler — APPROVED** [review]

**Pull request merged**